sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,417 @@
+                        row["nullable"] = False
+                    else:
+                        row["nullable"] = True
+                if table_id in id_to_table_data:
+                    id_to_table_data.get(table_id)["columns"] = id_to_table_data.get(table_id).get("columns", []) + [
+                        row
+                    ]
+                else:
+                    self._log.error("Columns found for an unkown table with the object_id: %s", table_id)
+            else:
+                self._log.error("Couldn't find id of a table: %s", table_id)
+        return len(data)
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_partitions_data(self, table_ids, id_to_table_data, cursor):
+        rows = execute_query_output_result_as_dicts(PARTITIONS_QUERY.format(table_ids), cursor)
+        for row in rows:
+            id = row.pop("id", None)
+            if id is not None:
+                id_str = str(id)
+                if id_str in id_to_table_data:
+                    id_to_table_data[id_str]["partitions"] = row
+                else:
+                    self._log.error("Partition found for an unkown table with the object_id: %s", id_str)
+            else:
+                self._log.error("Return rows of [%s] query should have id column", PARTITIONS_QUERY)
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_index_data(self, table_ids, id_to_table_data, cursor):
+        rows = execute_query_output_result_as_dicts(INDEX_QUERY.format(table_ids), cursor)
+        for row in rows:
+            id = row.pop("id", None)
+            if id is not None:
+                id_str = str(id)
+                if id_str in id_to_table_data:
+                    id_to_table_data[id_str].setdefault("indexes", [])
+                    id_to_table_data[id_str]["indexes"].append(row)
+                else:
+                    self._log.error("Index found for an unkown table with the object_id: %s", id_str)
+            else:
+                self._log.error("Return rows of [%s] query should have id column", INDEX_QUERY)
+
+    @tracked_method(agent_check_getter=agent_check_getter, track_result_length=True)
+    def _populate_with_foreign_keys_data(self, table_ids, id_to_table_data, cursor):
+        rows = execute_query_output_result_as_dicts(FOREIGN_KEY_QUERY.format(table_ids), cursor)
+        for row in rows:
+            id = row.pop("id", None)
+            if id is not None:
+                id_str = str(id)
+                if id_str in id_to_table_data:
+                    id_to_table_data.get(str(id)).setdefault("foreign_keys", [])
+                    id_to_table_data.get(str(id))["foreign_keys"].append(row)
+                else:
+                    self._log.error("Foreign key found for an unkown table with the object_id: %s", id_str)
+            else:
+                self._log.error("Return rows of [%s] query should have id column", FOREIGN_KEY_QUERY)
```

Comment:
and again https://github.com/DataDog/integrations-core/pull/17258/files#r1627971318

Reply:
thx